app/databse.py

```diff
@@ -0,0 +1,152 @@
+import os
+import uuid
+from datetime import datetime
+
+from sqlalchemy import create_engine, Column, String, ForeignKey, DateTime, Enum, Boolean, UUID
+from sqlalchemy.orm import sessionmaker, relationship, declarative_base
+from sqlalchemy_utils import database_exists, create_database, drop_database
+
+# DB variables
+DB_USER = os.environ.get("DB_USER")
+DB_PASSWORD = os.environ.get("DB_PASSWORD")
+DB_HOST = os.environ.get("DB_HOST")
+DB_PORT = os.environ.get("DB_PORT")
+DB_NAME = os.environ.get("DB_NAME", 'dogfinder')
+DB_PROTOCOL = os.environ.get("DB_PROTOCOL", "sqlite")
+
+
+# Set SQLite type adjustments (does not support UUID)
+def get_id_creator():
+    if DB_PROTOCOL == 'sqlite':
+        return lambda: str(uuid.uuid4())
+    else:
+        return uuid.uuid4
+
+
+def get_id_column_type():
+    if DB_PROTOCOL == 'sqlite':
+        return String
+    else:
+        return UUID(as_uuid=True)
+
+
+ID_COLUMN_TYPE = get_id_column_type()
+ID_CREATOR = get_id_creator()
+
+# DB field enums
+DOG_TYPE_ENUMS = Enum('lost', 'found', name='dog_type')
+DOG_SEX_ENUMS = Enum('male', 'female', name='dog_sex')
+
+# Create a base class for declarative models
+Base = declarative_base()
+
+
+class Dog(Base):
+    __tablename__ = 'dogs'
+
+    # Dog data
+    ## Entry data
+    id = Column(ID_COLUMN_TYPE, primary_key=True, default=ID_CREATOR)
```

Comment:
Can we just use autoincrement primary key here?

Reply:
That was my original choice, Tslil asked for UUIDs

Reply:
Why?

Reply:
From what I remember she said that:
1. They are unique over all the tables.
2. They don't indicate how many records we have.
I can't tag her here, so maybe you can take it up with her and you can see what you prefer.

Reply:
I'll talk to her. Thank you!